engine/src/main/java/org/terasology/particles/updating/ParticleUpdaterImpl.java

```diff
@@ -238,7 +318,7 @@ private void updateParticleSystem(final ParticleEmitterComponent partSys, final
             partSys.collisionUpdateIteration = (partSys.collisionUpdateIteration + 1) % PHYSICS_SKIP_NR;
         }
 
-        // System ran out of lifetime -> stop emission -> dispose
+        // System ran out of lifetime -> stop emission -> deregister
```

Comment:
why using a comment if you can just write `if(systemRanOutOfLifetime(...){deregister/unregister(...)}`?